larndsim/detsim_ep.py

```diff
@@ -291,11 +300,10 @@ def calc_total_current(self, x_start, y_start, z_start,
 
         z_step = (z_end_int - z_start_int) / (z_steps - 1)
 
-
         iz = ep.arange(z_steps, 0, z_steps.max().item())
         z =  z_start_int[:, :, ep.newaxis] + iz[ep.newaxis, ep.newaxis, :] * z_step[..., ep.newaxis]
 
-        t0 = (ep.abs(z - borders[:, 2, 0, ep.newaxis, ep.newaxis]) - 0.5) / self.vdrift
+        t0 = (ep.abs(z - borders[:, 2, 0, ep.newaxis, ep.newaxis]) - 0.5) / vdrift
```

Comment:
What's the explanation for this change?

Reply:
Torch checkpointing only keeps track of gradients for explicit inputs and outputs of the checkpointed function, so if you don't pass vdrift as an argument and vdrift has `requires_grad=True` it complains. Not the cleanest, but so it goes